torch/testing/_internal/subclasses.py

```diff
@@ -0,0 +1,76 @@
+# mypy: ignore-errors
+from typing import Any, Optional, Type
+
+import torch
+import torch.utils._pytree as pytree
+from torch.testing._internal.two_tensor import TwoTensor
+from torch.utils._python_dispatch import return_and_correct_aliasing
+
+
+class WrapSC(torch.Tensor):
+    @staticmethod
+    def __new__(cls, a, outer_size=None, outer_stride=None):
+        if outer_size is None:
+            outer_size = a.size()
+        if outer_stride is None:
+            outer_stride = a.stride()
+
+        kwargs = {}
+        kwargs["strides"] = a.stride()
+        kwargs["storage_offset"] = a.storage_offset()
+        kwargs["device"] = a.device
+        kwargs["layout"] = a.layout
+        kwargs["requires_grad"] = a.requires_grad
+        kwargs["dtype"] = a.dtype
+        out = torch.Tensor._make_wrapper_subclass(cls, a.size(), **kwargs)
+
+        return out
+
+    def __init__(self, a, outer_size=None, outer_stride=None):
+        self.a = a
+
+    def __repr__(self):
+        return f"WrapSC({repr(self.a)})"
+
+    def __tensor_flatten__(self):
+        return ["a"], None
+
+    @staticmethod
+    def __tensor_unflatten__(inner_tensors, meta, outer_size, outer_stride):
+        assert meta is None
+        a = inner_tensors["a"]
+        if type(a) is torch.Tensor:
```

Comment:
I guess the purpose of this check is that you only want to run the assertions below if we are at runtime, not trace time.

This isn't very robust though, since it will fail if the inner tensor `a` is itself another subclass.

You probably want to use `is_fake(a)`, to tell if you are in the middle of tracing?

Reply:
Honestly, I just copied this from TwoTensor :)